aws/aws-parallelcluster
# [DFSM] Allow dynamic file system mounting as live update on login nodes for external EFS/FSx* storage.

**This PR depends on https://github.com/aws/aws-parallelcluster/pull/6080**

### Description of changes
Allow dynamic file system mounting as live update on login nodes for external EFS/FSx* storage.
Added/revised unit tests.

Observations:
1. This PR is the equivalent of https://github.com/aws/aws-parallelcluster/pull/6060 for login nodes.

### Tests
* Dry-run update in the following cases:
  * add/remove external EFS/FSx: accepted as expected
  * add/remove managed EFS/FSx: rejected as expected
  * add/remove external/managed EBS: rejected as expected 
  * add/remove /home: rejected as expected
* Unit tests.

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
